apps/src/code-studio/pd/workshops/components/EnrollInWorkshop.tsx

```diff
@@ -124,31 +125,50 @@ const EnrollInWorkshop: React.FC<EnrollInWorkshopProps> = ({
       );
     }
 
+    const isMissingRequiredUserField =
+      !userInfo.first_name ||
+      !userInfo.last_name ||
+      !userInfo.email ||
+      !userInfo.school_info?.school_name;
     return (
-      <Button
-        className={moduleStyles.fullWidthButton}
-        type="primary"
-        size="m"
-        isPending={isSubmitting}
-        onClick={handleClick}
-        text="Enroll in this workshop"
-      />
+      <div className={moduleStyles.internalEnrollButton}>
+        {userInfo && (
+          <UserPassport
+            displayName={userInfo.display_name}
+            givenName={userInfo.first_name}
+            familyName={userInfo.last_name}
+            email={userInfo.email}
+            schoolName={userInfo.school_info?.school_name}
+            returnToHref={`/professional-learning/workshops/${id}`}
+            className={moduleStyles.userPassport}
+          />
+        )}
+        {alertState.show && (
+          <Alert
+            type={'danger'}
+            text={alertState.text}
+            link={alertState.link}
+            onClose={() =>
+              setAlertState({show: false, text: '', link: undefined})
+            }
+          />
+        )}
```

Comment:
Moved this section here to preserve the order shown in the Figma, which I believe is okay since this section wouldn't show in the full workshop view, the custom_registration_link workshop view, nor the signed-out/student view.